### PR TITLE
Update webhook notification workflow

### PR DESCRIPTION
**Pull request by BanzaiMan**
This PR documents [the new way](https://github.com/travis-ci/travis-tasks/pull/65) to verify the webhook notification's integrity using the private key signing.

Also adds a deprecation warning to the current way. The date is tentatively set to November 1, 2016.

### CODE REVIEW

File: user/notifications.md [hunk 1]
```diff
@@ -599,7 +599,29 @@ Here's a simple example of a [Sinatra](http://sinatrarb.com) app to decode the r
...
+
+To ensure the integrity of your workflow, we strongly encourage you to
+verify the POST request before acting on it.
+
+The POST request comes with the custom HTTP header `Signature`.
+Using the published SSL public key, you can verify the signature of the
+payload.
+
+1. Pick up the `payload` data from the HTTP request's body.
+1. Obtain the `Signature` header value.
```
Comment (aakritigupta) [inline user/notifications.md]: I think you means `2.` `3.`...

**Comment by aakritigupta** [comment]
Rest sounds good. :)

File: user/notifications.md (hunk 1 above)
Comment (BanzaiMan) [inline user/notifications.md]: I did mean `1.`, `1.`, `1.`. Markdown will renumber them correctly. https://daringfireball.net/projects/markdown/syntax#list

Comment (aakritigupta) [inline user/notifications.md]: Wow that's nice. TIL

File: user/notifications.md [hunk 2]
```diff
@@ -599,7 +599,29 @@ Here's a simple example of a [Sinatra](http://sinatrarb.com) app to decode the r
 
 To quickly identify the repository involved, we include a `Travis-Repo-Slug` header, with a format of `account/repository`, so for instance `travis-ci/travis-ci`.
 
-### Authorization for Webhooks
+### Verifying the Webhook requests
```
Comment (joshk) [inline user/notifications.md]: maybe remove the `the`

File: user/notifications.md [hunk 3]
```diff
@@ -599,7 +599,29 @@ Here's a simple example of a [Sinatra](http://sinatrarb.com) app to decode the r
...
+The POST request comes with the custom HTTP header `Signature`.
+Using the published SSL public key, you can verify the signature of the
+payload.
+
+1. Pick up the `payload` data from the HTTP request's body.
+1. Obtain the `Signature` header value.
+1. Obtain the public key corresponding to the private key that signed
+   the payload. This is available at the `/config` endpoint's
+   `config.notifications.webhook.public_key` on the relevant API server.
+1. Verify the signature using the public key and SHA1 digest.
```
Comment (joshk) [inline user/notifications.md]: doesn't the signature also need to be base64 decoded?

File: user/notifications.md [hunk 4]
```diff
@@ -599,7 +599,29 @@ Here's a simple example of a [Sinatra](http://sinatrarb.com) app to decode the r
...
+
+[WebhookSignatureVerifier](https://github.com/travis-ci/webhook-signature-verifier)
+is a small Sinatra app which shows you how this works.
+
+### Authorization for Webhooks (Deprecated)
+
+**Note: this method of authorizing webhooks will be deprecated on
```
Comment (joshk) [inline user/notifications.md]: maybe mention it is deprecated and will be removed on the 1st of Nov

File: user/notifications.md (hunk 3 above)
Comment (BanzaiMan) [inline user/notifications.md]: Yes, that is true!

**Pull request merged**